test/integration/test_collections.js

```diff
@@ -62,10 +64,32 @@ module.exports = {
          .set('Authorization', `Bearer ${TOKEN}`)
          .end((err, res) => {
            expect(err).to.be.null;
-           expect(res).to.have.status(200);
+           expect(res).to.have.status(statusCodes.SUCCESS);
            expect(res.text).to.equal('test1,test2 collection(s) deleted');
            done();
          });
+    },
+
+    'test_collection_upload': function(done) {
+      const test = exts => {
+        if (!exts.length) {
+          return done();
+        }
+
+        const ext = exts.shift();
+        chai.request(SERVER_URL)
+          .post(`${PATH_PREFIX}/collections/upload`)
+          .attach('file', fs.readFileSync(`${__dirname}/fixture/import.${ext}`), `import.${ext}` )
+          .set('Authorization', `Bearer ${TOKEN}`)
+          .end((err, res) => {
+            expect(err).to.be.null;
+            expect(res).to.have.status(statusCodes.CREATED);
+
+            mongodbClient.dropCollection('import', test.bind(null, exts));
+          });
+      };
+
+      test(['json','csv','bson']);
```

Comment:
I just want to make sure we are ok with using the recursive method here. It's slightly different to how other tests are written and we might want to maintain consistent code.

The other option is using async and would look something like
```javascript
    'test_collection_upload': function(done) {
      const test = (ext, cb) => {
        chai.request(SERVER_URL)
          .post(`${PATH_PREFIX}/collections/upload`)
          .attach('file', fs.readFileSync(`${__dirname}/fixture/import.${ext}`), `import.${ext}` )
          .set('Authorization', `Bearer ${TOKEN}`)
          .end((err, res) => {
            expect(err).to.be.null;
            expect(res).to.have.status(201);
            mongodbClient.dropCollection('import', cb);
          });
      };

      async.eachSeries(['json','csv','bson'],  test, done);
    }
```

Thoughts @pb82 @rachael-oregan ?